### PR TITLE
ux: improve API error handling and visibility

### PR DESCRIPTION
**Pull request by brunobergher**
## Motivation

API errors from different providers were handled inconsistently, and important error metadata (HTTP status codes, retry information) was being lost during error transformation. Additionally, error details were hidden behind a low-contrast icon that users often missed.

### Backend Error Handling
- **Unified error handler** in `error-handler.ts` standardizes error processing across all API providers (Anthropic, Gemini, Mistral, OpenAI)
- **Preserves critical metadata**: HTTP status codes, `errorDetails` (including RetryInfo for 429 errors), AWS metadata
- **Maintains backward compatibility** by keeping `openai-error-handler.ts` as a thin wrapper

### UX Improvements  
- **More prominent error details**: Replaced low-contrast info icon with inline text link that reads "View details"
- **Better error context**: Status codes and full error messages now flow through to `ErrorRow` for improved debugging
- **Clearer error attribution**: Instead of "API Error", we show "Provider Error", which is the real source of these.

Screenshot 2025-12-19 at 12 00 58

### Quality
- Added 469 new test cases covering error classification, metadata preservation, and edge cases
- All lint, type checks, and tests passing

----

> [!IMPORTANT]
> Improves API error handling by standardizing error processing and enhancing error visibility in the UI, with comprehensive testing.
> 
>   - **Backend Error Handling**:
>     - Introduces `handleProviderError` in `error-handler.ts` for unified error processing across Anthropic, Gemini, Mistral, OpenAI.
>     - Preserves HTTP status codes, `errorDetails`, AWS metadata.
>     - Maintains backward compatibility with `handleOpenAIError` as a wrapper.
>   - **UX Improvements**:
>     - Replaces low-contrast icon with inline text link "View details" for error visibility in `ErrorRow`.
>     - Displays status codes and full error messages in `ErrorRow`.
>     - Changes "API Error" to "Provider Error" for clearer attribution.
>   - **Quality**:
>     - Adds 469 test cases for error classification, metadata preservation, edge cases.
>     - All tests, lint, and type checks passing.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 294749ef32a21bca921e6dc0a15027d918258f95. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: webview-ui/src/i18n/locales/ja/chat.json [hunk 1]
```diff
@@ -264,6 +264,7 @@
 		"noToolsUsedDetails": "モデルはテキスト/推論を提供しましたが、必要なツールを呼び出しませんでした。これは通常、モデルがタスクを誤解したか、どのツールを使用するかを決定するのに苦労していることを示しています。モデルは適切なツールの使用で再試行するよう自動的に促されました。"
 	},
 	"errorDetails": {
+		"link": "Details",
```
Comment (ghost) [inline webview-ui/src/i18n/locales/ja/chat.json]: The `"link": "Details"` string is added in English across all 17 non-English locale files. For non-Latin script languages (Japanese, Chinese, Korean, Russian, Hindi, Vietnamese), this will display untranslated English text to users. Consider adding proper translations for each locale (e.g., "詳細" for Japanese, "详情" for Chinese, "세부정보" for Korean).

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=RooCodeInc/Roo-Code&prNumber=10204&fixId=417010e009) or mention @roomote and request a fix.</sub>

**Comment by ghost** [comment]
> **Rooviewer**  Clock &nbsp;&nbsp;[See task on Roo Cloud](https://app.roocode.com/cloud-jobs/UjRZYszT?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review)

Review complete. All previously identified issues have been resolved.

- [x] i18n translations for `errorDetails.link` properly added for non-Latin script languages (hi, ja, ko, ru, vi, zh-CN, zh-TW)

No new issues found. The PR implements:
- Unified error handler with proper metadata preservation (status, errorDetails, code, $metadata)
- UX improvement with more visible "View details" link
- Comprehensive test coverage (469 new test cases)
- Appropriate i18n updates across all supported locales

<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=RooCodeInc/Roo-Code&prNumber=10204&commentId=3674923515).</sub>
